tests/common/memwritestream.cpp

```diff
@@ -311,6 +311,70 @@ GTEST_TEST(MemoryWriteStream, writeString) {
 	EXPECT_EQ(data[5], 'r');
 }
 
+
+GTEST_TEST(MemoryWriteStream, seek) {
+	byte data[6] = { 0 };
+	Common::MemoryWriteStream stream(data);
+
+	stream.writeString("Foobar");
+	stream.seek(3);
+	stream.writeString("Foo");
+
+	EXPECT_EQ(data[0], 'F');
+	EXPECT_EQ(data[1], 'o');
+	EXPECT_EQ(data[2], 'o');
+	EXPECT_EQ(data[3], 'F');
+	EXPECT_EQ(data[4], 'o');
+	EXPECT_EQ(data[5], 'o');
+
+	stream.seek(-2, Common::SeekableWriteStream::kOriginEnd);
+	stream.writeString("uu");
+
+	EXPECT_EQ(data[0], 'F');
+	EXPECT_EQ(data[1], 'o');
+	EXPECT_EQ(data[2], 'o');
+	EXPECT_EQ(data[3], 'F');
+	EXPECT_EQ(data[4], 'u');
+	EXPECT_EQ(data[5], 'u');
+
+	stream.seek(-6, Common::SeekableWriteStream::kOriginCurrent);
+	stream.writeString("bar");
+
+	EXPECT_EQ(data[0], 'b');
+	EXPECT_EQ(data[1], 'a');
+	EXPECT_EQ(data[2], 'r');
+	EXPECT_EQ(data[3], 'F');
+	EXPECT_EQ(data[4], 'u');
+	EXPECT_EQ(data[5], 'u');
+}
+
+GTEST_TEST(MemoryWriteStream, size) {
+	byte data[14] = { 0 };
+	Common::MemoryWriteStream stream(data);
+
+	stream.writeUint64BE(0x0123456789ABCDEF);
+	EXPECT_EQ(stream.size(), 14);
+
+	stream.seek(-2, Common::SeekableWriteStream::kOriginCurrent);
+	stream.writeUint64BE(0x0123456789ABCDEF);
+	EXPECT_EQ(stream.size(), 14);
+
+	EXPECT_EQ(data[0], 0x01);
+	EXPECT_EQ(data[1], 0x23);
+	EXPECT_EQ(data[2], 0x45);
+	EXPECT_EQ(data[3], 0x67);
+	EXPECT_EQ(data[4], 0x89);
+	EXPECT_EQ(data[5], 0xAB);
+	EXPECT_EQ(data[6], 0x01);
+	EXPECT_EQ(data[7], 0x23);
+	EXPECT_EQ(data[8], 0x45);
+	EXPECT_EQ(data[9], 0x67);
+	EXPECT_EQ(data[10], 0x89);
+	EXPECT_EQ(data[11], 0xAB);
+	EXPECT_EQ(data[12], 0xCD);
+	EXPECT_EQ(data[13], 0xEF);
```

Comment:
I'm sorry, but I'm still not completely happy with this test.

It now doesn't test the case where the non-dynamic MemoryWriteStream has only space for less than the requested amount.

I.e. add another writing after this, in this test here:
```
static const byte kData[4] = { 0x12, 0x34, 0x56, 0x78 };
stream.seek(-2, Common::SeekableWriteStream::kOriginCurrent);
const size_t writeCount = stream.write(kData, 4);
EXPECT_EQ(writeCount, 2);
EXPECT_EQ(data[0], 0x01);
[...]
```

I.e. seek back, ask to write more than there's space for it, make sure only x bytes are written and compare to make sure the correct bytes are written.